src/main/java/de/tum/in/www1/artemis/service/scheduled/cache/monitoring/ExamMonitoringScheduleService.java

```diff
@@ -67,11 +73,37 @@ public static void configureHazelcast(Config config) {
         config.getScheduledExecutorConfig(Constants.HAZELCAST_MONITORING_SCHEDULER).setPoolSize(16).setCapacity(1000).setDurability(1);
     }
 
-    @EventListener(ApplicationReadyEvent.class)
-    public void applicationReady() {
-        // activate Exam Monitoring Service
-        SecurityUtils.setAuthorizationObject();
-        startSchedule();
+    /**
+     * This method schedules all exam activity save tasks after a server (re-)start.
+     */
+    @PostConstruct
+    public void startSchedule() {
+        try {
+            Collection<String> activeProfiles = Arrays.asList(env.getActiveProfiles());
+            if (activeProfiles.contains(JHipsterConstants.SPRING_PROFILE_DEVELOPMENT)) {
+                // only execute this on production server, i.e. when the prod profile is active
+                // NOTE: if you want to test this locally, please comment it out, but do not commit the changes
+                return;
+            }
+
+            if (!activeProfiles.contains("scheduling")) {
+                // only execute this on server with active scheduling profile
+                return;
+            }
+
+            SecurityUtils.setAuthorizationObject();
+
+            List<Exam> exams = examRepository.findAllCurrentAndUpcomingExams().stream().filter(Exam::isMonitoring).toList();
+            logger.info("Found {} exams that are not yet ended or are scheduled to start in the future", exams.size());
+            for (Exam exam : exams) {
+                if (exam.isMonitoring()) {
+                    scheduleExamMonitoringTask(exam.getId());
+                }
+            }
```

Comment:
You already filter above, you don't need to check this a second time.
```suggestion
            for (Exam exam : exams) {
                scheduleExamMonitoringTask(exam.getId());
            }
```

Reply:
This check is required since we have multiple entries for this method. See `StudentExamResource` -> `  instanceMessageSendService.sendExamMonitoringSchedule(examId);`